src/huggingface_hub/hf_api.py

```diff
@@ -244,11 +244,21 @@ class LastCommitInfo(TypedDict, total=False):
     date: datetime
 
 
-class BlobLfsInfo(TypedDict, total=False):
+@dataclass
+class BlobLfsInfo(dict):
     size: int
     sha256: str
     pointer_size: int
 
+    def __init__(self, data: dict):  # hack to make BlobLfsInfo backward compatible
+        self.__dict__.update(data)
+        self.size = data["size"]
+        self.sha256 = data["sha256"]
+        self.pointer_size = data["pointer_size"]
+
+    def __post_init__(self):
+        self.update(asdict(self))
```

Comment:
(no need of the `__init__` anymore)

```suggestion
    def __post_init__(self):  # hack to make BlobLfsInfo backward compatible
        self.update(asdict(self))
```